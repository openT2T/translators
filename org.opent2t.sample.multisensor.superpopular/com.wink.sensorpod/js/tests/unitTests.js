var translatorPath = require('path').join(__dirname, '..');
var runMultisensorTests = require('opent2t-device-multisensor/multisensorTests');
var deviceData = require('./devicedata');
var testData = require('./testdata');
var MockHub = require('opent2t-device-winkhub/mockWinkHub');
var mockHub = new MockHub(deviceData);

var settings = {
    translatorPath,
    getDeviceInfo: mockHub.getDeviceInfo.bind(mockHub),
    setTestData: mockHub.setTestData.bind(mockHub),
    expectedExceptions : testData.expected_exceptions,
    inputLastReading : deviceData.base_state.data.last_reading,
};

// Run standard lamp unit tests
runMultisensorTests(settings);