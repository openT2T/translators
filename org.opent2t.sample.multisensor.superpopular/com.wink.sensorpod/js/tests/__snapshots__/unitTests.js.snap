exports[`GetPlatform 1`] = `"{\"opent2t\":{\"schema\":\"org.opent2t.sample.multisensor.superpopular\",\"translator\":\"opent2t-translator-com-wink-sensorpod\",\"controlId\":\"123456\"},\"pi\":\"01234567-0123-0123-0123-0123456789ab\",\"mnmn\":\"linear\",\"mnmo\":\"linear_wapirz_1\",\"n\":\"Office Motion Sensor\",\"rt\":[\"org.opent2t.sample.multisensor.superpopular\"],\"entities\":[{\"n\":\"Office Motion Sensor\",\"rt\":[\"opent2t.d.sensor.brightnesschange\"],\"di\":\"11dd1be4-b699-162f-f545-39cc93cd9e3f\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/brightnesschange\",\"rt\":[\"oic.r.sensor\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]},{\"href\":\"/lastchanged\",\"rt\":[\"opent2t.r.timestamp\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]}]},{\"n\":\"Office Motion Sensor\",\"rt\":[\"opent2t.d.sensor.contact\"],\"di\":\"1fdde8d7-0cee-3b24-878e-bed7f271fd8a\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/contact\",\"rt\":[\"oic.r.sensor.contact\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]},{\"href\":\"/lastchanged\",\"rt\":[\"opent2t.r.timestamp\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]}]},{\"n\":\"Office Motion Sensor\",\"rt\":[\"opent2t.d.sensor.humidity\"],\"di\":\"70e7841e-6240-f9b6-c407-6150c62f2ea1\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/humidity\",\"rt\":[\"oic.r.humidity\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]},{\"href\":\"/lastchanged\",\"rt\":[\"opent2t.r.timestamp\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]}]},{\"n\":\"Office Motion Sensor\",\"rt\":[\"opent2t.d.sensor.locked\"],\"di\":\"03057ad8-2379-81df-3300-123e0e01facb\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/locked\",\"rt\":[\"oic.r.sensor\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]},{\"href\":\"/lastchanged\",\"rt\":[\"opent2t.r.timestamp\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]}]},{\"n\":\"Office Motion Sensor\",\"rt\":[\"opent2t.d.sensor.loudnesschange\"],\"di\":\"3837331e-1e55-774d-00a9-092055f646d1\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/loudnesschange\",\"rt\":[\"oic.r.sensor\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]},{\"href\":\"/lastchanged\",\"rt\":[\"opent2t.r.timestamp\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]}]},{\"n\":\"Office Motion Sensor\",\"rt\":[\"opent2t.d.sensor.motion\"],\"di\":\"23e8af9b-af94-0dc2-5bd3-f4059b20953c\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/motion\",\"rt\":[\"oic.r.sensor.motion\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]},{\"href\":\"/lastchanged\",\"rt\":[\"opent2t.r.timestamp\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]}]},{\"n\":\"Office Motion Sensor\",\"rt\":[\"opent2t.d.sensor.presence\"],\"di\":\"52554766-d7ba-4cb2-e031-e3e3a105227a\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/presence\",\"rt\":[\"oic.r.sensor.presence\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]},{\"href\":\"/lastchanged\",\"rt\":[\"opent2t.r.timestamp\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]}]},{\"n\":\"Office Motion Sensor\",\"rt\":[\"opent2t.d.sensor.temperature\"],\"di\":\"77aba2a2-6155-5e56-3179-32c4d59d31f4\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/temperature\",\"rt\":[\"oic.r.temperature\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]},{\"href\":\"/lastchanged\",\"rt\":[\"opent2t.r.timestamp\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]}]},{\"n\":\"Office Motion Sensor\",\"rt\":[\"opent2t.d.sensor.vibrationchange\"],\"di\":\"da5575a6-68d0-b1b1-c798-107e8ca2ea30\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/vibrationchange\",\"rt\":[\"oic.r.sensor\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]},{\"href\":\"/lastchanged\",\"rt\":[\"opent2t.r.timestamp\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]}]},{\"n\":\"Office Motion Sensor\",\"rt\":[\"opent2t.d.sensor.water\"],\"di\":\"ea47bc4c-699d-797d-f3b6-38a8b69f37c0\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/water\",\"rt\":[\"oic.r.sensor.water\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]},{\"href\":\"/lastchanged\",\"rt\":[\"opent2t.r.timestamp\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]}]},{\"n\":\"Office Motion Sensor\",\"rt\":[\"opent2t.d.battery\"],\"di\":\"3fb458a2-5f01-9117-11c2-c0f154c2b351\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/battery\",\"rt\":[\"oic.r.energy.battery\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"]}]}]}"`;

exports[`GetPlatformExpanded 1`] = `"{\"opent2t\":{\"schema\":\"org.opent2t.sample.multisensor.superpopular\",\"translator\":\"opent2t-translator-com-wink-sensorpod\",\"controlId\":\"123456\"},\"pi\":\"01234567-0123-0123-0123-0123456789ab\",\"mnmn\":\"linear\",\"mnmo\":\"linear_wapirz_1\",\"n\":\"Office Motion Sensor\",\"rt\":[\"org.opent2t.sample.multisensor.superpopular\"],\"entities\":[{\"n\":\"Office Motion Sensor\",\"rt\":[\"opent2t.d.sensor.brightnesschange\"],\"di\":\"11dd1be4-b699-162f-f545-39cc93cd9e3f\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/brightnesschange\",\"rt\":[\"oic.r.sensor\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"],\"id\":\"brightnesschange\",\"value\":false},{\"href\":\"/lastchanged\",\"rt\":[\"opent2t.r.timestamp\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"],\"id\":\"lastchanged\",\"timestamp\":\"2017-03-07T00:36:40.374Z\"}]},{\"n\":\"Office Motion Sensor\",\"rt\":[\"opent2t.d.sensor.contact\"],\"di\":\"1fdde8d7-0cee-3b24-878e-bed7f271fd8a\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/contact\",\"rt\":[\"oic.r.sensor.contact\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"],\"id\":\"contact\",\"value\":true},{\"href\":\"/lastchanged\",\"rt\":[\"opent2t.r.timestamp\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"],\"id\":\"lastchanged\",\"timestamp\":\"2017-03-07T00:36:40.374Z\"}]},{\"n\":\"Office Motion Sensor\",\"rt\":[\"opent2t.d.sensor.humidity\"],\"di\":\"70e7841e-6240-f9b6-c407-6150c62f2ea1\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/humidity\",\"rt\":[\"oic.r.humidity\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"],\"id\":\"humidity\",\"humidity\":50},{\"href\":\"/lastchanged\",\"rt\":[\"opent2t.r.timestamp\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"],\"id\":\"lastchanged\",\"timestamp\":\"2017-03-07T00:36:40.374Z\"}]},{\"n\":\"Office Motion Sensor\",\"rt\":[\"opent2t.d.sensor.locked\"],\"di\":\"03057ad8-2379-81df-3300-123e0e01facb\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/locked\",\"rt\":[\"oic.r.sensor\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"],\"id\":\"locked\",\"value\":false},{\"href\":\"/lastchanged\",\"rt\":[\"opent2t.r.timestamp\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"],\"id\":\"lastchanged\",\"timestamp\":\"2017-03-07T00:36:40.374Z\"}]},{\"n\":\"Office Motion Sensor\",\"rt\":[\"opent2t.d.sensor.loudnesschange\"],\"di\":\"3837331e-1e55-774d-00a9-092055f646d1\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/loudnesschange\",\"rt\":[\"oic.r.sensor\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"],\"id\":\"loudnesschange\",\"value\":false},{\"href\":\"/lastchanged\",\"rt\":[\"opent2t.r.timestamp\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"],\"id\":\"lastchanged\",\"timestamp\":\"2017-03-07T00:36:40.374Z\"}]},{\"n\":\"Office Motion Sensor\",\"rt\":[\"opent2t.d.sensor.motion\"],\"di\":\"23e8af9b-af94-0dc2-5bd3-f4059b20953c\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/motion\",\"rt\":[\"oic.r.sensor.motion\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"],\"id\":\"motion\",\"value\":true},{\"href\":\"/lastchanged\",\"rt\":[\"opent2t.r.timestamp\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"],\"id\":\"lastchanged\",\"timestamp\":\"2017-03-07T00:36:40.374Z\"}]},{\"n\":\"Office Motion Sensor\",\"rt\":[\"opent2t.d.sensor.presence\"],\"di\":\"52554766-d7ba-4cb2-e031-e3e3a105227a\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/presence\",\"rt\":[\"oic.r.sensor.presence\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"],\"id\":\"presence\",\"value\":false},{\"href\":\"/lastchanged\",\"rt\":[\"opent2t.r.timestamp\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"],\"id\":\"lastchanged\",\"timestamp\":\"2017-03-07T00:36:40.374Z\"}]},{\"n\":\"Office Motion Sensor\",\"rt\":[\"opent2t.d.sensor.temperature\"],\"di\":\"77aba2a2-6155-5e56-3179-32c4d59d31f4\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/temperature\",\"rt\":[\"oic.r.temperature\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"],\"id\":\"temperature\",\"temperature\":22.77777777777778,\"units\":\"C\"},{\"href\":\"/lastchanged\",\"rt\":[\"opent2t.r.timestamp\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"],\"id\":\"lastchanged\",\"timestamp\":\"2017-03-07T00:42:16.508Z\"}]},{\"n\":\"Office Motion Sensor\",\"rt\":[\"opent2t.d.sensor.vibrationchange\"],\"di\":\"da5575a6-68d0-b1b1-c798-107e8ca2ea30\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/vibrationchange\",\"rt\":[\"oic.r.sensor\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"],\"id\":\"vibrationchange\",\"value\":true},{\"href\":\"/lastchanged\",\"rt\":[\"opent2t.r.timestamp\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"],\"id\":\"lastchanged\",\"timestamp\":\"2017-03-07T00:36:40.374Z\"}]},{\"n\":\"Office Motion Sensor\",\"rt\":[\"opent2t.d.sensor.water\"],\"di\":\"ea47bc4c-699d-797d-f3b6-38a8b69f37c0\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/water\",\"rt\":[\"oic.r.sensor.water\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"],\"id\":\"water\",\"value\":false},{\"href\":\"/lastchanged\",\"rt\":[\"opent2t.r.timestamp\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"],\"id\":\"lastchanged\",\"timestamp\":\"2017-03-07T00:36:40.374Z\"}]},{\"n\":\"Office Motion Sensor\",\"rt\":[\"opent2t.d.battery\"],\"di\":\"3fb458a2-5f01-9117-11c2-c0f154c2b351\",\"icv\":\"core.1.1.0\",\"dmv\":\"res.1.1.0\",\"resources\":[{\"href\":\"/battery\",\"rt\":[\"oic.r.energy.battery\"],\"if\":[\"oic.if.s\",\"oic.if.baseline\"],\"id\":\"battery\",\"charge\":0}]}]}"`;
